src/main/java/pl/edu/mimuw/logic/Solver.java

```diff
@@ -0,0 +1,65 @@
+package pl.edu.mimuw.logic;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class Solver {
+  private static Boolean[] valToBoolean(Valuation val, ArrayList<Variable> variables) {
+    Boolean[] res = new Boolean[variables.size()];
+    int i = 0;
+
+    for (Variable variable : variables)
+      res[i++] = val.get(variable.getName());
+
+    return res;
+  }
+
+  private static Valuation valFromBoolean(Boolean[] tried, ArrayList<Variable> variables) {
+    Valuation gen = new Valuation();
+    int i = 0;
+
+    for (Variable variable : variables)
+      gen.add(variable.getName(), tried[i++]);
+
+    return gen;
+  }
+
+  private static Valuation allFalseValuation(ArrayList<Variable> variables) {
+    Valuation gen = new Valuation();
+    for (Variable var : variables)
+      gen.add(var.getName(), false);
+    return gen;
+  }
+
+  public static Valuation solveBrutally(Formula formula) {
+    ArrayList<Variable> variables = new ArrayList<>(formula.getVariables());
+    Queue<Valuation> queue = new ConcurrentLinkedQueue<>();
```

Comment:
zapewne chodziło Ci o `LinkedList<>()`, które też implementuje interfejs kolejski, bądź też o `ArrayDequeue`. To czego użyłeś służy do wykorzystanie w programach współbieżnych, to za rok chyba będzie 😉 (a przez to jest wolniejsze, bo bezpieczniejsze)